driver/controller.go

```diff
@@ -209,6 +211,83 @@ func (s controllerService) DeleteVolume(ctx context.Context, req *csi.DeleteVolu
 	return &csi.DeleteVolumeResponse{}, nil
 }
 
+// CreateSnapshot creates a logical volume snapshot.
+func (s controllerService) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	var (
+		// Since the kubernetes snapshots are Read-Only, we set accessType as 'ro' to activate thin-snapshots as read-only volumes
```

Comment:
Since for us, both snapshots and restored volumes are thin-snapshots in the backend; we differentiate them by activating them with different permissions. 
Since we can have two possible accessTypes for snapshots and restores, we need to set that variable, as In logical_volume_controller, we call the `createlv` method for creating any type of logical volume(snapshot or regular); so there it doesn't know if the request has come from `CreateVolume` gRPC or `CreateSnapshot` gRPC. 
[See](https://github.com/topolvm/topolvm/blob/main/controllers/logicalvolume_controller.go#L87)

 It makes its calling decisions based on the values it gets from the logicalvolume CR.
Thus, to differentiate between a snapshot and a restored volume(so we can activate it differently in lvm), we need to specify the `AccessType` parameter while creating the LogicalVolume CR.
[See](https://github.com/topolvm/topolvm/pull/463/commits/03755940baa3cae1f0448e006992c645a36271c0#diff-7b9e88d2ad217bf9b768f9a41e9d4e857ce02ff9b66d0032138c573626d3459bR214)